subprojects/GOPATH/src/github.com/microsoft/hivedscheduler/pkg/algorithm/hived_algorithm.go

```diff
@@ -800,16 +807,19 @@ func generatePodScheduleResult(
 		// may cause the selected node to be excluded from the suggested nodes
 		// TODO: Keep selectedNode within suggestedNodeSet, so here should be Unreachable
 		if !suggestedNodeSet.Contains(selectedNode) && newGroup {
-			panic(fmt.Sprintf("[%v]: node %v picked by algorithm but not in K8S candidates",
-				internal.Key(pod), selectedNode))
+			return internal.PodScheduleResult{
+				PodWaitInfo: &internal.PodWaitInfo{
+					Reason: fmt.Sprintf("node %v picked by algorithm but not in K8S candidates", selectedNode),
+				},
+			}
```

Comment:
Let's keep this still be panic.
Because after you keep selectedNode within suggestedNodeSet, so here should be Unreachable, and if it really reaches, it is exception (code bug).

Reply:
I mean we keep allNodesInSuggested in getFewestOpporPhysicalCell, etc

---
In reply to: [353628498](https://github.com/microsoft/pai/pull/3962#discussion_r353628498) [](ancestors = 353628498)

Reply:
I mean, we can't do this. Some nodes may be excluded from the the suggested nodes only because there are other pods on it (we need to preempt them). If we only consider suggested nodes when scheduling, such preemption will never happen

---
In reply to: [353649575](https://github.com/microsoft/pai/pull/3962#discussion_r353649575) [](ancestors = 353649575,353628498)